mesonbuild/compilers/compilers.py

```diff
@@ -287,6 +287,7 @@ class CompileCheckMode(enum.Enum):
     OptionKey('b_pgo'): coredata.UserComboOption('Use profile guided optimization',
                                                  ['off', 'generate', 'use'],
                                                  'off'),
+    OptionKey('b_profile'): coredata.UserBooleanOption('Generate extra code to write profile information suitable for gprof', False),
```

Comment:
The documentation above documents this as '[off, prof, gprof]' but here you define it as a boolean option.